software/base/src/main/java/org/apache/brooklyn/entity/software/base/TargetServerEntityImpl.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.entity.software.base;
+
+import java.util.Collection;
+
+import org.apache.brooklyn.api.entity.Entity;
+import org.apache.brooklyn.api.location.Location;
+import org.apache.brooklyn.core.entity.Entities;
+import org.apache.brooklyn.core.entity.EntityInternal;
+
+public class TargetServerEntityImpl extends VanillaSoftwareProcessImpl implements TargetServerEntity {
+
+    protected void preStart() {
+        Entity target = config().get(TARGET_ENTITY);
+        Entities.waitForServiceUp(target);
+        Collection<Location> locations = target.getLocations();
+        clearLocations();
```

Comment:
Would be surprising to have locations defined on the entity. Could log a warning if there are any.